quickshop-bukkit/src/main/java/com/ghostchu/quickshop/menu/ShopTradeMenu.java

```diff
@@ -30,7 +32,9 @@ public class ShopTradeMenu extends QuickShopMenu {
 
   public ShopTradeMenu() {
 
-    this.rows = 5;
+    // Load rows from config or use default (6 rows for modern layout)
+    final GuiConfig.MenuConfig menuConfig = QuickShop.getInstance().getGuiConfig().getMenuConfig("trade");
+    this.rows = menuConfig != null ? menuConfig.getRows() : 6;
```

Comment:
The ternary condition should have parentheses around it, with the question mark operator having no space before it.